tests/e2e/csi_static_provisioning_file_basic.go

```diff
@@ -0,0 +1,239 @@
+	"github.com/onsi/gomega"
+	"github.com/vmware/govmomi/cns"
+	cnsmethods "github.com/vmware/govmomi/cns/methods"
+	cnstypes "github.com/vmware/govmomi/cns/types"
+	vsanfstypes "github.com/vmware/govmomi/vsan/vsanfs/types"
+	v1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	fnodes "k8s.io/kubernetes/test/e2e/framework/node"
+	fpod "k8s.io/kubernetes/test/e2e/framework/pod"
+	fpv "k8s.io/kubernetes/test/e2e/framework/pv"
+)
+
+var _ = ginkgo.Describe("[csi-file-vanilla] Basic File Volume Static Provisioning", func() {
+	f := framework.NewDefaultFramework("e2e-csifilestaticprovision")
+
+	var (
+		client            clientset.Interface
+		datastoreURL      string
+		defaultDatacenter *object.Datacenter
+		defaultDatastore  *object.Datastore
+		namespace         string
+		pv                *v1.PersistentVolume
+		pvc               *v1.PersistentVolumeClaim
+	)
+
+	ginkgo.BeforeEach(func() {
+		bootstrap()
+		client = f.ClientSet
+		namespace = f.Namespace.Name
+		nodeList, err := fnodes.GetReadySchedulableNodes(f.ClientSet)
+		framework.ExpectNoError(err, "Unable to find ready and schedulable Node")
+		if !(len(nodeList.Items) > 0) {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+
+		var datacenters []string
+		datastoreURL = GetAndExpectStringEnvVar(envSharedDatastoreURL)
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		finder := find.NewFinder(e2eVSphere.Client.Client, false)
+		cfg, err := getConfig()
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		dcList := strings.Split(cfg.Global.Datacenters, ",")
+		for _, dc := range dcList {
+			dcName := strings.TrimSpace(dc)
+			if dcName != "" {
+				datacenters = append(datacenters, dcName)
+			}
+		}
+
+		for _, dc := range datacenters {
+			defaultDatacenter, err = finder.Datacenter(ctx, dc)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			finder.SetDatacenter(defaultDatacenter)
+			defaultDatastore, err = getDatastoreByURL(ctx, datastoreURL, defaultDatacenter)
+			if err == nil {
+				break
+			}
+		}
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Datastore is not found in the datacenter list")
+	})
+
+	/*
+		This test verifies the static provisioning workflow.
+
+		Test Steps:
+		1. Create File share and get the file share id.
+		2. Create PV Spec with volumeID set to file share ID created in Step-1, and PersistentVolumeReclaimPolicy is set to Delete.
+		3. Create PVC with the storage request set to PV's storage capacity.
+		4. Wait for PV and PVC to bound.
+		5. Create a POD.
+		6. Verify volume is attached to the node and volume is accessible in the pod by creating a file inside volume.
+		7. Verify container volume metadata is present in CNS cache.
+		8. Delete POD.
+		9. Delete PVC.
+		10. Verify PV is deleted automatically.
+	*/
+	ginkgo.It("Verify basic static provisioning workflow", func() {
```

Comment:
Can you clarify that this is for file volumes?

Reply:
Done.